vcs/git/repo.go

```diff
@@ -235,11 +233,19 @@ func (r *Repository) Commits(opt vcs.CommitsOptions) ([]*vcs.Commit, uint, error
 			commits = append(commits, r.makeCommit(c))
 		}
 		total++
-		return total < uint(MaxCommits)
```

Comment:
As discussed, there are no known `git` backend users (and if there are and care about this, they can start using `NoTotal` option), so let's side code that is simpler and produces correct (per documentation) results.